httpx/models.py

```diff
@@ -209,7 +209,19 @@ def __str__(self) -> str:
 
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
-        url_str = str(self)
```

Comment:
Let's use a proper block `if ... else ...` here. I think it's too opaque otherwise.